### PR TITLE
Add Dave2 Executor (Issue 83)

### PR DESCRIPTION
**Pull request by p1ndsvin**

### CODE REVIEW

File: competition.py [hunk 1]
```diff
@@ -214,7 +216,7 @@ def log_exception(extype, value, trace):
               show_default='Disabled',
               help="Activate debugging (results in more logging)")
 @click.pass_context
-def generate(ctx, executor, beamng_home, beamng_user,
+def generate(ctx, executor, ai_path, beamng_home, beamng_user,
```
Comment (alessiogambi) [inline competition.py]: @p1ndsvin What's `ai_path` ?

File: self_driving/nvidia_prediction.py [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+            image = np.array([image])
+
+            steering_angle = float(self.model.predict(image, batch_size=1))
+
+            speed = car_state.vel_kmh
+            if speed > self.speed_limit:
+                self.speed_limit = MIN_SPEED  # slow down
+            else:
+                self.speed_limit = MAX_SPEED
+            throttle = 1.0 - steering_angle ** 2 - (speed / self.speed_limit) ** 2
```
Comment (alessiogambi) [inline self_driving/nvidia_prediction.py:28]: @p1ndsvin Do we configure the `speed_limit` using the input parameter passed to `competition.py`?

File: self_driving/vehicle_state_reader.py [hunk 3]
```diff
@@ -36,6 +36,7 @@ def __init__(self, vehicle: Vehicle, beamng: BeamNGpy, additional_sensors: List[
         self.vehicle.attach_sensor('electrics', electrics)
         self.vehicle.attach_sensor('timer', timer)
 
+        #TODO
```
Comment (alessiogambi) [inline self_driving/vehicle_state_reader.py]: Probably we should get rid of this one...

**Review by alessiogambi — COMMENTED** [review]
I still need to test it, and there are a couple of clarifications.

File: competition.py (hunk 1 above)
Comment (p1ndsvin) [reply competition.py]: check the doc/help if it's clear

File: self_driving/nvidia_prediction.py (hunk 2 above)
Comment (p1ndsvin) [reply self_driving/nvidia_prediction.py:28]: the parameter can substitute MAX_SPEED

File: self_driving/vehicle_state_reader.py (hunk 3 above)
Comment (p1ndsvin) [reply self_driving/vehicle_state_reader.py]: probably yes

**Comment by alessiogambi** [comment]
TODO: 

- Improve input validation
- Add FAQ for tensorflow related problems: https://github.com/tensorflow/tensorflow/issues/35749

**Comment by alessiogambi** [comment]
@p1ndsvin I tested manually and it seems to work. Please check that it works for you as well and then you can merge the pull request and close this issue.

**Pull request merged**